jubilant/_juju.py

```diff
@@ -703,6 +703,36 @@ def run(
             if params_file is not None:
                 os.remove(params_file.name)
 
+    def scp(
+        self,
+        source: str,
```

Comment:
Idea -- since `source` without any special characters `@`, `:` is just the path bit, it would be simple to support `pathlib.Path` here, which could be nice as some users (me) pretty much always use `pathlib.Path` for local file paths ... though currently users would just need to write `source=str(path)`

```suggestion
        source: str | pathlib.Path,
```

`args.append(source)` on line 731 would also need to become `args.append(str(source))`

Reply:
Good call. And same for `destination`, right? (See the integration tests, where destination is a local path.)

Reply:
Or maybe `os.PathLike[str]` like `deploy`'s *charm* argument?

Reply:
> same for destination, right?

Oh if `destination` has the same semantics then yes.

I thought from the help text that it didn't -- for `source` 'foo' would be treated as a local file named foo, while for `destination` 'foo' would be treated as the ssh target -- so `destination=Path('foo')` being treated as the ssh target would be surprising if we did `args.append(str(destination))` -- though maybe there's a way to tell Juju it's a local path in that case?

Reply:
> Or maybe os.PathLike[str] like deploy's charm argument?

That would work too, though I believe that the right way to get a string representation of a filesystem path from an [os.PathLike object](https://docs.python.org/3/library/os.html#os.PathLike) is via its `__fspath__` method, while `pathlib.Path` is both `os.PathLike` and guarantees that `str` is the string representation of the filesystem path

Reply:
Cool, I've updated this (and other similar places) to `pathlib.Path`. I've also added it to `refresh()`'s *path* argument.